### ISSUE TITLE
训练过程中提示Variable value (label) of OP(fluid.layers.cross_entropy) expected >= 0 and < 1, but got 9

### ISSUE THREAD

**Issue by GentleCP** [PaddlePaddle/Paddle#24307]
在用官网的aistudio环境训练时出现未知问题
- 版本、环境信息：
    1）PaddlePaddle版本：1.7.1
    2）CPU：aistudio 基础版
    4）系统环境：aistudio环境
- 训练信息
    1）单机
    2）显存信息
    3）Operator信息
- 问题描述：请详细描述您的问题，同步贴出报错信息、日志、可复现的代码片段

Error Message Summary:
----------------------
Error: Variable value (label) of OP(fluid.layers.cross_entropy) expected >= 0 and < 1, but got 9. Please check label value. at (/paddle/paddle/fluid/operators/cross_entropy_op.h:173)

**Comment by cryoco** [comment]
您好，看报错信息是在cross_entropy中，label的值超过了类别数的上限。
可以参考cross_entropy的文档，https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/cross_entropy_cn.html#cross-entropy
，input的最后一维是类别数目，在您的报错信息中显示您把它设为了1，但却出现了值为9的label，请检查下这里是否设置有误哈～

**Comment by GentleCP** [comment]
> 您好，看报错信息是在cross_entropy中，label的值超过了类别数的上限。
> 可以参考cross_entropy的文档，https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/cross_entropy_cn.html#cross-entropy
> ，input的最后一维是类别数目，在您的报错信息中显示您把它设为了1，但却出现了值为9的label，请检查下这里是否设置有误哈～

哦找到了，是我网络输出那里类别数写错了,感谢感谢。 但是训练的时候出现了新的问题，我的predict是[128,10]的(128是batch_size)，label是[128,1] 。通过交叉熵计算得到的loss data是nan，不知道是什么原因。下面是predict 和label的部分内容：
- predict  
name tmp_13, dtype: VarType.FP32 shape: [128, 10] 	lod: {}
	dim: 128, 10
	layout: NCHW
	dtype: float
	data: [-1.66276 -1.82948 -1.29232 0.713727 -1.42557 -1.37891 0.168096 0.953176 0.412122 0.0383092 -0.784453 -0.883333 -1.06902 0.423694 -0.942533 -0.627647 0.540182 0.222206...
- label 
name generated_var_1, dtype: VarType.INT64 shape: [128, 1] 	lod: {}
	dim: 128, 1
	layout: NCHW
	dtype: int64_t
	data: [9 1 2 0 0 6 4 7 6 5 3 7 8 1 4 7 1 5 5 0 3 6 4 1 0 7 9 6 9 6 6 7 3 3 3 2 3 4 5 
- loss 
epoch: 0, batch: 0, loss is: [nan]

- 我的项目地址是: https://aistudio.baidu.com/aistudio/projectdetail/450595
> 可以的话能帮我看看是什么问题吗，感谢

**Comment by cryoco** [comment]
您好，请问问题是否解决了哈

**Comment by GentleCP** [comment]
是的已经解决啦，感谢。之前是网络输出的问题 574881148 Email:574881148@qq.com Signature is customized by Netease Mail Master 在2020年05月06日 10:52，Pei Yang 写道： 您好，请问问题是否解决了哈 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by moyang2024** [comment]
能问一下，在哪更改label的值吗

**Comment by GentleCP** [comment]
> 能问一下，在哪更改label的值吗

fluid.layers.fc(input=conv_pool,size=10,act='softmax'), 在你的网络输出最后一层，全连接出去的size和act修改

**Comment by moyang2024** [comment]
> > 能问一下，在哪更改label的值吗
> 
> fluid.layers.fc(input=conv_pool,size=10,act='softmax'), 在你的网络输出最后一层，全连接出去的size和act修改
def convolutional_neural_network(image, type_size):

predict = fluid.layers.fc(input=drop,size=type_size,act='softmax')

predict = convolutional_neural_network(image=image,type_size=9)

**Comment by moyang2024** [comment]
我传的参数这样设置应该没问题了，可还是提示label大于4了

**Comment by GentleCP** [comment]
> 我传的参数这样设置应该没问题了，可还是提示label大于4了

这要根据你的情况定了，我是10分类，所以size=10，激活函数用`softmax`，你需要注意你自己的分类数量

**Comment by moyang2024** [comment]
嗯，谢谢

**Comment by hellokhj** [comment]
> 您好，看报错信息是在cross_entropy中，label的值超过了类别数的上限。 可以参考cross_entropy的文档，https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/cross_entropy_cn.html#cross-entropy ，input的最后一维是类别数目，在您的报错信息中显示您把它设为了1，但却出现了值为9的label，请检查下这里是否设置有误哈～

请问在哪里修改
